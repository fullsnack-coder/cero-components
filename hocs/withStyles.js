import React from 'react'
import { getClasses } from '@glrodasz/storybook-tools-styles'

const withStyles = (styles) => (WrappedComponent) => {
  const WithStylesComponent = (props) => {
    const allProps = { ...WrappedComponent.defaultProps, ...props }
    return (
      <WrappedComponent getStyles={getClasses(styles)(allProps)} {...props} />
    )
  }

  WithStylesComponent.displayName = WrappedComponent.displayName

  return WithStylesComponent
}

export default withStyles
