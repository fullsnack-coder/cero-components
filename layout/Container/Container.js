import React from 'react'
import PropTypes from 'prop-types'
import styles from './Container.module.css'
import withStyles from '../../hocs/withStyles'

export const Container = ({
  getStyles,
  children,
  isCentered,
  isPlayground,
}) => {
  return (
    <div
      className={getStyles('container', {
        'is-playground': isPlayground,
        'is-centered': isCentered,
      })}
    >
      {children}
    </div>
  )
}

Container.propTypes = {
  children: PropTypes.node.isRequired,
  getStyles: PropTypes.func.isRequired,
  isPlayground: PropTypes.bool,
  isCentered: PropTypes.bool,
}

Container.defaultProps = {
  isPlayground: false,
  isCentered: true,
  getStyles: () => {},
}

export default withStyles(styles)(Container)
