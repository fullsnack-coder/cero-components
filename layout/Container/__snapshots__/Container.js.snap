// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ storybook ] [ Layout/Container ] should render Centered 1`] = `
<DocumentFragment>
  <div
    class="container is-centered"
  />
</DocumentFragment>
`;

exports[`[ storybook ] [ Layout/Container ] should render Default 1`] = `
<DocumentFragment>
  <div
    class="container"
  />
</DocumentFragment>
`;
